pkg/scheduler/objects/application_test.go

```diff
@@ -2600,6 +2600,39 @@ func TestGetRateLimitedAppLog(t *testing.T) {
 	assert.Check(t, l != nil)
 }
 
+func TestTryAllocateWithReservedHeadRoomChecking(t *testing.T) {
+	var res, headRoom *resources.Resource
+	var err error
+	res, err = resources.NewResourceFromConf(map[string]string{"memory": "2G"})
+	assert.NilError(t, err, "failed to create basic resource")
+	headRoom, err = resources.NewResourceFromConf(map[string]string{"memory": "1G"})
+	assert.NilError(t, err, "failed to create basic resource")
+
+	app := newApplication(appID1, "default", "root")
+
+	ask := newAllocationAsk(aKey, appID1, res)
+
+	queue, err := createRootQueue(map[string]string{"memory": "1G"})
+	assert.NilError(t, err, "queue create failed")
+	app.queue = queue
+
+	err = app.AddAllocationAsk(ask)
+	assert.NilError(t, err, "ask should have been added to app")
+
+	node1 := newNodeRes(nodeID1, res)
+	node2 := newNodeRes(nodeID2, res)
+	// reserve that works
+	err = app.Reserve(node1, ask)
+	assert.NilError(t, err, "reservation should not have failed")
+	if app.IsReservedOnNode("") {
+		t.Errorf("app should not have reservations for empty node ID")
+	}
+
```

Comment:
Add a recover call to catch any regressions:
```
	defer func() {
		if r := recover(); r != nil {
			t.Fatalf("reserved headroom test regression: %v", r)
		}
	}()
```
If the DPanic causes a panic while running the test we will properly catch it and cause the test to fail with a clear message that can be actioned.

Reply:
Thanks again for the further review Wilfred, good to learn this. I added it in the new rev and it worked nicely.